File: GasturaApp/Core/Entities/Gasto.cs
using GasturaApp.Core.Enums;
using System.ComponentModel.DataAnnotations;

namespace GasturaApp.Core.Entities;

public class Gasto
{
    [Key]
    public int Id { get; set; }

    [Required]
    public decimal Valor { get; set; }

    public DateTime? DataHora { get; set; }

    public string Descricao { get; set; } = string.Empty;

    [Required]
    public ModalidadePagamento ModalidadePagamento { get; set; }

    [Required]
    public int UsuarioId { get; set; }

    [Required]
    public int CategoriaId { get; set; }

    public Usuario Usuario { get; set; }
    public Categoria Categoria { get; set; }
}

File: gastura-app/src/types/Gasto.ts
import Categoria from "./Categoria";
import ModalidadePagamento from "./ModalidadePagamento";
import Usuario from "./Usuario";

interface Gasto {
  id: string;
  valor: number;
  dataHora?: Date;
  descricao: string;
  modalidadePagamento: ModalidadePagamento;
  usuarioId: string;
  categoriaId: string;
  categoria: Categoria | null;
  usuario: Usuario;
}

export default Gasto;